From e83a4f7b53cb8ca14db6983428aca7dbf0acd565 Mon Sep 17 00:00:00 2001
From: "Mazlan, Hazwan Arif" <hazwan.arif.mazlan@intel.com>
Date: Fri, 20 May 2022 15:33:09 +0800
Subject: [PATCH 20/22] i915/gt/uc: override GuC firmware for ADLN

override GuC path to load tgl GuC like ADLS for ADLN as a temporary
WA while waiting for official fix patch

Signed-off-by: James Xiong <james.xiong@intel.com>
Signed-off-by: Mazlan, Hazwan Arif <hazwan.arif.mazlan@intel.com>
---
 drivers/gpu/drm/i915/gt/uc/intel_uc_fw.c | 9 +++++++++
 1 file changed, 9 insertions(+)

diff --git a/drivers/gpu/drm/i915/gt/uc/intel_uc_fw.c b/drivers/gpu/drm/i915/gt/uc/intel_uc_fw.c
index 54f161b39fe5..c80cee19d2e7 100644
--- a/drivers/gpu/drm/i915/gt/uc/intel_uc_fw.c
+++ b/drivers/gpu/drm/i915/gt/uc/intel_uc_fw.c
@@ -217,6 +217,15 @@ static void __uc_fw_user_override(struct drm_i915_private *i915, struct intel_uc
 	switch (uc_fw->type) {
 	case INTEL_UC_FW_TYPE_GUC:
 		path = __override_guc_firmware_path(i915);
+
+		if (path == NULL && IS_ADLP_N(i915)) {
+		   static const struct uc_fw_blob adln_blob = GUC_FW_BLOB(tgl, 70, 0, 3);
+		   uc_fw->path = adln_blob.path;
+		   uc_fw->major_ver_wanted = adln_blob.major;
+		   uc_fw->minor_ver_wanted = adln_blob.minor;
+		   uc_fw->user_overridden = true;
+		}
+
 		break;
 	case INTEL_UC_FW_TYPE_HUC:
 		path = __override_huc_firmware_path(i915);
-- 
2.25.1

